internal/pkg/override/cdk.go

```diff
@@ -172,6 +182,92 @@ func (cdk *CDK) cleanUp(in []byte) ([]byte, error) {
 	return out.Bytes(), nil
 }
 
+type packageManager struct {
+	name     string
+	lockFile string
+}
+
+var packageManagers = []packageManager{ // Alphabetically sorted based on name.
+	{
+		name:     "npm",
```

Comment:
can we use constants for `npm` and `yarn`
```
const (
    npm   = "npm"
    yarn  = "yarn"
)
```

Reply:
I prefer keeping it as is if that's fine with you! Here is my reasoning -

I feel like we want to use `constant`s (instead of a "magic value") mostly in code logic:
```go
// Example: "npm" is a magic value
if _, err := cdk.exec.LookPath("npm"); err == nil {}

// Example: no magic value
npm := "npm"
if _, err := cdk.exec.LookPath(npm); err == nil {}
```

Using constant instead magic value makes it easier to refactor the code - you just need to change the value of `npm` to take care of all references to it. 

In this case, the `packageManagers` is a package-scoped variable instead of a part of "the logic". Consider this - if we use constants here, the code will looks like:
```go
const (
    npm   = "npm"
    yarn  = "yarn"
)

var packageManagers = []packageManager{
   {  name: npm, }
   {   name: yarn, }
}

if _, err := cdk.exec.LookPath(candidate.name); err == nil {}

```

The number of magic number did not decrease. All the logic code is still using `candidate.name` to refer to the package manager name. The only place that refers to `npm` is `packageManagers`.

Simply put, I feel like the problems with using "magic values" is already solved by `candidate.name` - and adding a layer of constant on top of `packageManagers` may not be a huge help.

Reply:
But I did add this https://github.com/aws/copilot-cli/pull/5076/commits/7ad6de58ca07f4aa43640b4ea5f711e750e7093e that's to address your comment!